### DISCUSSION TITLE
need to get ready for react 18

### DISCUSSION THREAD

**Discussion by OleksiiKachan** [pmndrs/react-spring#1562] [Support]
https://github.com/reactwg/react-18/discussions/41

react 18 will have useTransition hook. need to get ready for this collision.

I'd suggest to rename react-spring useTransition(s) hooks to something else (I have no suggestions so far)
the sooner we will have a solution for this, the sooner users can upgrade and prepare for react 18

**Comment by joshuaellis** [comment]
You're welcome to submit a PR, although I think this is a bigger issue and needs to be scoped accordingly.

**Reply by OleksiiKachan** [reply]
as this is a breaking change, I'd like to see what community thinks first, before trying to change anything. react 18 release is not tomorrow, so
we have time to discuss this issue

**Reply by joshuaellis** [reply]
In fact, i'm going to make an issue and lock this one so theres no confusion. thanks for the prod.
